lib/Fetcher/FeedFetcher.php

```diff
@@ -344,6 +344,30 @@ protected function buildItem(
         return $item;
     }
 
+    /**
+     * Helper function for fetching favicon urls
+     *
+     * @param string        $base_url  Base URL for the feed
+     *
+     * @return string|mixed|bool
+     */
+    protected function faviconHelper(string $base_url)
+    {
+        try {
+            return $this->faviconFactory->get($base_url);
+        } catch (\Error $e) {
+            //Actual error should be ValueError but that is only implemented in PHP 8+
```

Comment:
If it's only in PHP 8+, how can the factory use it? Aren't they using their own `ValueError`? Because otherwise it can't throw it either.

Reply:
Yea I think it's not their own ValueError, the lib doesn't define any Exceptions if I'm not wrong instead this error from the issue came from PHP itself. Which I guess depends on your PHP version.

The docs for the function that is throwing the error did not inlcude further information.

This should catch the error in php 8+ which the report was also based on and below that worst case the error is not caught because they used something else..

Reply:
Can we reproduce it somehow so I can trace and update it in the lib?

Reply:
I don't know, the linked issue did have some hints on how it probably happens but for that you would need to miss misconfigure a Webserver on purpose, with an invalid redirect for the favicon.

Reply:
It would be easier if we just find the feed where this happens. If we don't make this library more resiliant to errors we'll keep running into this.

Reply:
True but the reporter was apparently not able to name the affected feed. The feed that was mentioned there did not cause that error.